tests/php/test_class.json-api-jetpack-endpoints.php

```diff
@@ -16,25 +14,35 @@ private function set_globals() {
 	}
 
 	public function setUp() {
+		if ( ! defined( 'WPCOM_JSON_API__BASE' ) ) {
+			define( 'WPCOM_JSON_API__BASE', 'public-api.wordpress.com/rest/v1' );
+		}
 
 		parent::setUp();
 
+		if ( version_compare( PHP_VERSION, '5.3', '<' ) ) {
+			// Loading the API breaks in 5.2.x due to `const` declarations, and
+			// this will still happen even though all tests are skipped (why?)
+			return;
```

Comment:
because of static analysis I guess. An approach that we took with one specific test was making the check for the version and then conditionally including/requiring a file with the new syntax that would be unsupported in php 5.2 for example.

Reply:
Oh I've just seen the full file and looks like here we're doing the same. Maybe I trolled with my comment.

Reply:
For reference I was talking about this changeset https://github.com/Automattic/jetpack/pull/8790

Reply:
Here is my theory on this file - not confirmed, but I dont think it's worth spending much more effort.

Before, this entire file was marked as `@group external-http`, and phpunit was smart enough to skip the whole thing.

After, we have some tests that are `@group json-api`, and even though all tests in the file require PHP >= 5.3, phpunit still runs the `setUp` method because it doesn't realize that all tests will be skipped.

So we need to abort the `setUp` method as well, and avoid PHP 5.2 trying to load the API files that use `const`.